test/fixtures/es-module-loaders/hook-resolve-type.mjs

```diff
@@ -1,6 +1,28 @@
 let importedESM = 0;
 let importedCJS = 0;
-global.getModuleTypeStats = () => { return {importedESM, importedCJS} };
+
+export function globalPreload({ port }) {
+  port.on('message', (int32) => {
+    port.postMessage({ importedESM, importedCJS });
+    Atomics.store(int32, 0, 1);
+    Atomics.notify(int32, 0);
+  });
+  port.unref();
+  return `
+    const { receiveMessageOnPort } = getBuiltin('worker_threads');
+    global.getModuleTypeStats = async function getModuleTypeStats() {
+      const sab = new SharedArrayBuffer(4);
+      const int32 = new Int32Array(sab);
+      port.postMessage(int32);
+      // Artificial timeout to keep the event loop alive.
+      // https://bugs.chromium.org/p/v8/issues/detail?id=13238
+      // TODO(targos) Remove when V8 issue is resolved.
+      setTimeout(() => {}, 100);
+      await Atomics.waitAsync(int32, 0, 0).value;
+      return receiveMessageOnPort(port).message;
```

Comment:
/cc @joyeecheung this is related to https://bugs.chromium.org/p/v8/issues/detail?id=13238 / https://github.com/nodejs/node/pull/44409